### PR TITLE
Add Qwen4Exp multimodal pipeline

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- add the Qwen3.8 Flash-Next/Qwen4Exp three-model image-language package (`decoder`, `vision_encoder`, `embedding`)
- reuse the source-identical no-DeepStack Qwen3/Qwen3.5 vision tower with a 2560-wide merger
- preserve independent lexical `ple_input_ids`, fused `inputs_embeds`, four-channel text + M-RoPE positions, and every heterogeneous recurrent/KV/QSA/PLE state
- cast processor float32 patches once at the BF16 vision boundary
- emit explicit `mobius.state_manifest` and `mobius.unsupported_token_ids` contracts for direct ONNX Runtime orchestration

## Bounded multimodal streaming

Both `qwen4_exp` and `qwen4_exp_text` use the bounded safetensors path; the eager `_download_weights` route is unreachable. One shard index validates config-derived deterministic PLE buffers and plans decoder, embedding, and vision bindings. Each graph is cloned, lazily bound, and initializer-folded in isolation; the package is updated only after every staged component succeeds. Missing tensors, corrupt PLE hash buffers, or fold failures leave the original package untouched.

Parameter payloads remain `LazyTensor`s and no source state dict is retained. The approximately 95 GiB PLE table is allocated once only when materialized/serialized and populated one checkpoint shard at a time. Loader accounting reports model/initializer/lazy counts, deterministic bytes validated eagerly, and zero retained source tensors.

## Enforced image-only boundary

The checkpoint's source video token is validated, then removed from normalized runtime metadata. Explicit configs requesting video capability fail closed and the embedding graph exposes no `video_features` input. It sanitizes any source video token before vocabulary lookup, computes a runtime `has_video`, and reshapes one zero element to `[1 + has_video]`. Valid requests reduce/add zero; video requests demand two elements and fail ONNX Reshape's element-count invariant before an EP executes. ORT CUDA uses the shared `ReshapeHelper` for this check (`onnxruntime@b1f76d58`, `cuda/tensor/reshape.h`), avoiding CUDA Gather's out-of-range zero-fill behavior. Optimized CPU is exercised locally and the same test executes CUDA when available; processor-native mixed input also fails through the guard.

## Runtime workflow boundary

onnxruntime-genai and the current ONNX GenAI workflow schema cannot represent Qwen4Exp's `ple_input_ids`, four-axis positions, and heterogeneous state membership. Both exporters reject before writing artifacts. ONNX GenAI detection checks package config, explicit config, and the structural decoder ABI, so a non-Qwen override cannot bypass preflight or modify an existing output directory. Direct ONNX Runtime users consume `mobius.state_manifest` instead.

## Source pins

- `Qwen/Qwen3.8-Flash-Next@f5d08274bafd880402bd16f5e3e6c514136ec06c`
- `huggingface/transformers@598d8ba8baaec7fec5a22da0e2844c7bf4ea20e1`

## Final-base rebase

Lower-layer PR #656 was squash-merged to `main` as `8f902ac2`. This branch replays only #657's multimodal work onto that commit; no pre-squash lower-layer or GGUF implementation files remain in the PR diff.

## Validation

- final Qwen4 loader/multimodal/synthetic suite: 47 passed, 6 skipped
- full runtime exporter suites: 263 passed
- pinned HuggingFace image parity plus real image/video boundary tests: 3 passed
- post-rebase GGUF report/shard/runtime regression suite: 831 passed
- Qwen4 component inspection and review-thread regressions: passed
- pinned lintrunner format + lint: clean
- independent GPT-5.6 Sol CUDA-guard rereview: no findings

## Waivers

- **L4/L5 real-weight golden:** the 512-expert checkpoint requires a large GPU; the immutable YAML case records the CI skip.
- **Released runtime workflows:** ORT GenAI and ONNX GenAI metadata export fail closed as described above.
- **Full checkpoint CUDA generation:** checkpoint scale makes this impractical in this session; CUDA's shared Reshape validation is source-pinned and the guard regression runs on CUDA whenever that EP is available.

This PR targets `main`, is intentionally non-draft, and must not be merged by this session.

### CODE REVIEW

File: src/mobius/models/qwen4_exp.py [hunk 1]
```diff
@@ -1261,3 +1289,114 @@ def preprocess_weights(
...
+    def forward(
+        self,
+        op: OpBuilder,
+        inputs_embeds: ir.Value,
+        ple_input_ids: ir.Value,
+        attention_mask: ir.Value,
+        position_ids: ir.Value,
+        past_position_ids: ir.Value,
+        past_key_values: list[tuple[ir.Value, ...]],
+    ):
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/qwen4_exp.py:1335]: Fixed in `5313deb9`. `Qwen4ExpVLDecoderModel.forward` now matches the base positional contract and accepts `inputs_embeds` / `ple_input_ids` as optional keyword-only inputs, with explicit validation when lexical IDs are unavailable.

File: src/mobius/tasks/_qwen4_exp.py [hunk 2]
```diff
@@ -5,13 +5,211 @@
...
+
+def _make_qwen4_state_inputs(
+    builder,
+    config: Qwen4ExpConfig,
+    batch: ir.SymbolicDim,
+    past_seq_len: ir.SymbolicDim,
+    past_position_ids: ir.Value,
+) -> list[tuple[ir.Value, ...]]:
+    """Create every Qwen4-Exp recurrent, PLE, QSA, and KV state input."""
+    assert config.layer_types is not None
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/tasks/_qwen4_exp.py]: Fixed in `5313deb9`. All three task paths now use `_require_layer_types`, which raises `ValueError` instead of relying on optimization-strippable assertions.

File: src/mobius/models/__init__.py [hunk 3]
```diff
@@ -374,7 +375,12 @@
     Qwen3TTSCodecEncoderModel,
     Qwen3TTSTokenizerV2Model,
 )
-from mobius.models.qwen4_exp import Qwen4ExpCausalLMModel
+from mobius.models.qwen4_exp import (
+    Qwen4ExpCausalLMModel,
+)
+from mobius.models.qwen4_exp import (
+    Qwen4ExpForConditionalGeneration as Qwen4ExpForConditionalGeneration,
+)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/__init__.py:381]: Fixed in `5313deb9`. The adjacent imports are consolidated into one normal import block; lint remains clean.
